test/integration/reasoner/planner/PlannerTest.java

```diff
@@ -180,12 +206,42 @@ public void test_planner_disconnected_conjunction() {
         Concludable concludable2 = Concludable.create(resolvedConjunction("{ $c($d); }", logicMgr)).iterator().next();
 
         Set<Resolvable<?>> resolvables = set(concludable, concludable2);
-        List<Resolvable<?>> plan = Planner.plan(resolvables, new HashMap<>(), set());
+        List<Resolvable<?>> plan = ReasonerPlannerTestWrapper.planResolvables(resolvables, set());
 
         assertEquals(2, plan.size());
         assertEquals(set(concludable, concludable2), set(plan));
     }
 
+    @Test
+    public void test_planner_negations_with_dependencies() {
+        transaction.query().define(TypeQL.parseQuery("define employment sub relation, relates employee, relates employer;" +
+                "work-permit sub relation, relates person;" +
+                "person plays employment:employee, plays work-permit:person;"));
+        transaction.commit();
+        session.close();
+        initialise(Arguments.Session.Type.DATA, Arguments.Transaction.Type.READ);
+
+        EntityType person = conceptMgr.putEntityType("person");
+        AttributeType name = conceptMgr.putAttributeType("name", AttributeType.ValueType.STRING);
+        person.setOwns(name);
+        conceptMgr.putRelationType("employment");
+        conceptMgr.putRelationType("work-permit");
+
+
+        Conjunction conjunction = resolvedConjunction("{ ($p,$e) isa employment;  not{ ($p) isa work-permit; }; }", logicMgr);
+        Set<Concludable> concludables = Concludable.create(conjunction);
+        Negated negated = new Negated(conjunction.negations().get(0));
+
+        Set<Resolvable<?>> resolvables = new HashSet<>();
+        resolvables.add(negated);
+        resolvables.addAll(concludables);
+
+        List<Resolvable<?>> plan = ReasonerPlannerTestWrapper.planResolvables(resolvables, set());
+
+        assertEquals(plan.size(), 2);
+        assertEquals(plan.get(1), negated);
+    }
+
```

Comment:
Add a case to ensure the negation (as failure) is evaluated only after the unnegated dependency.